src/components/checkbox/Checkbox.base.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+import Shape from '../../foundations/shape';
+import Grid from '../../foundations/layout';
+import Icon, { TIconSize } from '../icon';
+import Text, { TTextSizeToken } from '../text';
+import { Utils } from '../../shared';
+
+import PCheckbox from './Checkbox.props';
+import { DEFAULT_CHECKBOX_SIZE } from './Checkbox.constants';
+
+const CheckboxBase: React.FC<PCheckbox> = ({
+    label = 'Public',
+    size = DEFAULT_CHECKBOX_SIZE,
+    className,
+    onClick,
+    checked,
+}: PCheckbox): JSX.Element => {
+    let labelSize: TTextSizeToken = 100;
+    let iconSize: TIconSize = 12;
+    let checkboxSize: TCheckboxSize = 16;
+
+    const hasLabel = Utils.isString(label) && label.length > 0;
+
+    switch (size) {
```

Comment:
IMO we can better use dedicated maps (defined in `Checkbox.constants.ts`) that map the sizeToken to a specific value, when there is no computation needed.

usage example:
```typescript
<Icon glyph="check" size={CHECKBOX_ICON_SIZE_MAP[size]} />
```

That way we have a single point of truth for those values.